Marlin/Marlin_main.cpp

```diff
@@ -9363,14 +9363,19 @@ void set_current_from_steppers_for_axis(const AxisEnum axis) {
       // For non-interpolated delta calculate every segment
       for (uint16_t s = segments + 1; --s;) {
         DELTA_NEXT(segment_distance[i]);
-        planner.buffer_line_kinematic(DELTA_VAR, _feedrate_mm_s, active_extruder);
+        DELTA_IK();
+        ADJUST_DELTA(DELTA_VAR); // Adjust Z if bed leveling is enabled
+        planner.buffer_line(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], DELTA_VAR[E_AXIS], _feedrate_mm_s, active_extruder);
```

Comment:
Here `buffer_line_kinematic` does what we want. Since `USE_DELTA_IK_INTERPOLATION` is disabled in this case, there's no need to do the delta handling here. The `buffer_line_kinematic` function calls `planner.apply_leveling` which does what `ADJUST_DELTA` does.